openslides_backend/action/mixins/import_mixins.py

```diff
@@ -177,5 +180,51 @@ def validate_instance(self, instance: Dict[str, Any]) -> None:
                             raise ActionException(
                                 f"Could not parse {entry[field]} expect boolean"
                             )
+                    elif type_ == "date":
+                        try:
+                            entry[field] = int(
+                                mktime(strptime(entry[field], "%Y-%m-%d"))
+                            )
+                        except Exception:
+                            raise ActionException(
+                                f"Could not parse {entry[field]} except date"
+                            )
+                    elif type_ == "string[]":
+                        try:
+                            entry[field] = json.loads("[" + entry[field] + "]")
+                        except Exception:
+                            raise ActionException(
+                                f"Could not parse {entry[field]} except string[]"
+                            )
 
         super().validate_instance(instance)
+
+
+class Lookup:
+    def __init__(
+        self,
+        datastore: DatastoreService,
+        collection: str,
+        names: List[str],
+        field: str = "name",
+    ) -> None:
+        if not names:
+            self.name_to_id: Dict[str, Optional[int]] = {}
+        else:
+            self.name_to_id = {
+                entry[field]: entry["id"]
+                for entry in datastore.filter(
+                    collection,
+                    Or(*[FilterOperator(field, "=", name) for name in names]),
+                    ["id", field],
+                ).values()
+            }
+
+    def check_duplicate(self, name: str) -> bool:
```

Comment:
Here you have the cases
1. name not in dict
2. name in dict, but empty list
3. name in dict with one element in list
4. name in dict with more than one element
The signification of the 4 cases depends on the object:
For committee case 1 should initiate a create committee, case 3 an update, the others are errors: case 2 committee-name more than 1 time in import, case 4: update not possible, because found more than 1 committee
For meeting_name in committee import it doesn't matter, because we will always import, if there is a name
The meeting_template search must have only 1 id for the name, otherwise gives warning
... 

bool is no good return type. You could write specific return values for the different object (types) or give more information to this one, what to do with the 4 cases

Reply:
I return a `ResultType` enum. So the 4 cases could be handled.